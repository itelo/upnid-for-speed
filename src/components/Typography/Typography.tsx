import React from "react";
import styled from "styled-components";
import { motion, MotionProps } from "framer-motion";

type TypographyProps = {
  family: "FasterOne" | "Monofett" | "PressStart2P";
  color: "yellow" | "blue";
  children: React.ReactNode;
} & MotionProps &
  React.HTMLAttributes<HTMLParagraphElement>;

const TypographyStyled = styled(motion.p)<TypographyProps>`
  ${({ family }) => {
    if (family === "FasterOne") {
      return 'font-family: "Faster One", cursive;';
    }
    if (family === "Monofett") {
      return 'font-family: "Monofett", cursive;';
    }

    return 'font-family: "Press Start 2P", cursive;';
  }}
  ${({ color }) => {
    if (color === "blue") {
      return "color: rgb(86, 117, 184);";
    } else {
      return "color: rgb(234, 199, 133);";
    }
  }}
  
  font-size: 4.5rem;
  text-align: center;
  margin: 0;
`;

const Typography = (props: TypographyProps) => {
  return <TypographyStyled {...props} />;
};

Typography.defaultProps = {
  family: 'FasterOne',
  color: 'yellow'
};

export default Typography;
