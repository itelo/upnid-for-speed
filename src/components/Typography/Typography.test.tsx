import React from "react";
import { render, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import Typography from "./Typography";
import "jest-styled-components";

describe("<Typography />", () => {
  it("test basic render without passing any props", async () => {
    const { container, getByTestId } = render(
      <Typography data-testid="text">hello</Typography>
    );

    expect(getByTestId("text")).toHaveTextContent("hello");
    expect(container.firstChild).toMatchSnapshot();
  });

  it("test basic render with family='FasterOne'", async () => {
    const { container, getByTestId } = render(
      <Typography family="FasterOne" data-testid="text">
        hello
      </Typography>
    );

    expect(getByTestId("text")).toHaveTextContent("hello");
    // @ts-ignore
    expect(getByTestId("text")).toHaveStyleRule(
      "font-family",
      '"Faster One",cursive'
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("test basic render with family='Monofett'", async () => {
    const { container, getByTestId } = render(
      <Typography family="Monofett" data-testid="text">
        hello
      </Typography>
    );

    expect(getByTestId("text")).toHaveTextContent("hello");
    // @ts-ignore
    expect(getByTestId("text")).toHaveStyleRule(
      "font-family",
      '"Monofett",cursive'
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("test basic render with family='PressStart2P'", async () => {
    const { container, getByTestId } = render(
      <Typography family="PressStart2P" data-testid="text">
        hello
      </Typography>
    );

    expect(getByTestId("text")).toHaveTextContent("hello");
    // @ts-ignore
    expect(getByTestId("text")).toHaveStyleRule(
      "font-family",
      '"Press Start 2P",cursive'
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("test basic render with color='blue'", async () => {
    const { container } = render(
      <Typography color='blue' data-testid="text">
        hello
      </Typography>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("test basic render with color='yellow'", async () => {
    const { container, getByTestId } = render(
      <Typography color='yellow' data-testid="text">
        hello
      </Typography>
    );

    expect(getByTestId("text")).toHaveTextContent("hello");
    expect(container.firstChild).toMatchSnapshot();
  });
});


