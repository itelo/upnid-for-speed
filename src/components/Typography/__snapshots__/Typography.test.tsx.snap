// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography /> test basic render with color='blue' 1`] = `
.c0 {
  font-family: "Faster One",cursive;
  color: rgb(86,117,184);
  font-size: 4.5rem;
  text-align: center;
  margin: 0;
}

<p
  class="c0"
  color="blue"
  data-testid="text"
>
  hello
</p>
`;

exports[`<Typography /> test basic render with color='yellow' 1`] = `
.c0 {
  font-family: "Faster One",cursive;
  color: rgb(234,199,133);
  font-size: 4.5rem;
  text-align: center;
  margin: 0;
}

<p
  class="c0"
  color="yellow"
  data-testid="text"
>
  hello
</p>
`;

exports[`<Typography /> test basic render with family='FasterOne' 1`] = `
.c0 {
  font-family: "Faster One",cursive;
  color: rgb(234,199,133);
  font-size: 4.5rem;
  text-align: center;
  margin: 0;
}

<p
  class="c0"
  color="yellow"
  data-testid="text"
>
  hello
</p>
`;

exports[`<Typography /> test basic render with family='Monofett' 1`] = `
.c0 {
  font-family: "Monofett",cursive;
  color: rgb(234,199,133);
  font-size: 4.5rem;
  text-align: center;
  margin: 0;
}

<p
  class="c0"
  color="yellow"
  data-testid="text"
>
  hello
</p>
`;

exports[`<Typography /> test basic render with family='PressStart2P' 1`] = `
.c0 {
  font-family: "Press Start 2P",cursive;
  color: rgb(234,199,133);
  font-size: 4.5rem;
  text-align: center;
  margin: 0;
}

<p
  class="c0"
  color="yellow"
  data-testid="text"
>
  hello
</p>
`;

exports[`<Typography /> test basic render without passing any props 1`] = `
.c0 {
  font-family: "Faster One",cursive;
  color: rgb(234,199,133);
  font-size: 4.5rem;
  text-align: center;
  margin: 0;
}

<p
  class="c0"
  color="yellow"
  data-testid="text"
>
  hello
</p>
`;
